### ISSUE TITLE
experimentalEditContextEnabled breaks markdown.editor.pasteUrlAsFormattedLink.enabled

### ISSUE THREAD

**Issue by carlocardella** [microsoft/vscode#231707]
Does this issue occur when all extensions are disabled?: **Yes**

I can repro with the latest insiders and Stable builds:

```
Version: 1.95.0-insider
Commit: c85acdad6b679973c79b01a38afa91448c705c1c
Date: 2024-10-18T07:31:27.077Z
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Code-Insiders/1.95.0-insider Chrome/128.0.6613.162 Electron/32.1.2 Safari/537.36
```

Steps to Reproduce:

- This is today's Insiders build, no extensions, no custom settings
- Create a markdown file and paste a URL: the _paste as..._ picker is shown as expected
- Add/Enable `"editor.experimentalEditContextEnabled": true`
- Paste the link again: this time the _paste as..._ picker is not shown and it is not possible to force it with other keyboard shortcuts
- Pasting an image in the markdown file is also broken the same way: the image is simply not pasted, nothing happens

![Image](https://github.com/user-attachments/assets/cdfe2df2-91b6-4d4f-942d-55f6af897168)

**Comment by mjbvz** [comment]
One future alternative would be to switch to use the [browser async clipboard api](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) however this does not have the same functionality as the events right now. For example we currently  can't write out important mime types such as `text/uri-list`. Until those limitations are addressed I don't think we can switch to use it

Maybe we could try to create a paste on a hidden textarea so that we still get a real event? Here's what I'm thinking:

- cmd+v is detected as a paste by our keybindings
- We create a text area that is not visible to the user. This textarea also would also be outside of the edit context
- We use  `execCommand` to trigger a real paste event on that hidden textarea
- We then dispatch this paste event back to any listeners. The paste event object should remain valid for the synchronous dispatch of the event

**Comment by aiday-mar** [comment]
Hi @mjbvz thank you for the suggestion I have the following PR which fixes this issue : https://github.com/microsoft/vscode/pull/232806. The PR however introduces a new method `getTextAreaDomNode` into the ICodeEditor interface to solve this. I am not quite sure this is a good idea, and I wanted to ask you for a review.

**Comment by carlocardella** [comment]
The paste problem is fixed bug I think this fix introduced a different problem, see #233012
